third_party/rules_pycross/pycross/private/tools/wheel_installer.py

```diff
@@ -1,19 +1,35 @@
+# Copyright 2023 Jeremy Volkman. All rights reserved.
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 """
 A tool that invokes pypa/build to build the given sdist tarball.
 """
 
+import argparse
 import os
 import shutil
+import sys
 import tempfile
 from pathlib import Path
 from typing import Any
 
-from absl import app
-from absl.flags import argparse_flags
 from installer import install
 from installer.destinations import SchemeDictionaryDestination
 from installer.sources import WheelFile
-from pycross.private.tools import namespace_pkgs
+
+from rules_python.python.pip_install.tools.wheel_installer import namespace_pkgs
```

Comment:
Hmm. Do you have a link that documents the runfiles layout change with bzlmod? I was under the impression that it affected how external dependencies are managed, not how they are laid out in the runfiles tree.

Reply:
I think the layout doesn't change, it's just the fact that there is no folder in the runfiles tree named `rules_python` it is called something like `rules_python~extra_things_here` and we don't have anything in `rules_python` to use the repository mapping information to do some magic to create top level import structure. I am not sure if it is even possible to do so.

Reply:
Hmm. Looks like this does indeed not work, you're right. However, the runfiles layout did change. The runfiles used to be laid out in directories for each repository. Which is technically still true, but with unknowable names. The runfiles API compensates for this by reading in a `_repo_mapping` file. But Python obviously won't look at that.

Hmm. That means bzlmod will break a lot of stuff I have set up internally.

Either way, will switch to just `python.pip_install.tools.wheel_installer`. Though I'm pretty sure that will not work with `--experimental_python_import_all_repositories=false`.

Reply:
Done

Reply:
I think it should work with that flag because the `py_library` `imports` for the `runfiles` target is populated correctly, so the `PyInfo` provider will propage the imports and it should continue to work with `--experimental_python_import_all_repositories=false`.